temporalio/worker/workflow_sandbox/_restrictions.py

```diff
@@ -943,26 +952,37 @@ def r_op(obj: Any, other: Any) -> Any:
 
 def _is_restrictable(v: Any) -> bool:
     return v is not None and not isinstance(
-        v, (bool, int, float, complex, str, bytes, bytearray)
+        v, (bool, int, float, complex, str, bytes, bytearray, datetime)
```

Comment:
TODO: As things stand this means that users will be able to do `my_datetime.now()`. That's not the normal way to call `datetime.now()`. One option is to allow this only when the pydantic data converter is being used / pydantic is importable.

If we get rid of this change and allow `datetime` instances to be restricted, then Pydantic validation fails. This failure is in Pydantic's Rust layer: https://github.com/pydantic/pydantic-core/blob/main/src/errors/types.rs#L706-L707

```
datetime_field
  Input should be a valid datetime [type=datetime_type, input_value=datetime.datetime(2000, 1, 2, 3, 4, 5), input_type=_RestrictedProxy]
    For further information visit https://errors.pydantic.dev/2.10/v/datetime_type
```